byzcoin/viewchange.go

```diff
@@ -169,38 +170,40 @@ func (s *Service) sendNewView(proof []viewchange.InitReq) {
 		}
 	}
 
-	sb := s.db().GetByID(proof[0].View.ID)
-	req := viewchange.NewViewReq{
-		Roster: *rotateRoster(sb.Roster, proof[0].View.LeaderIndex),
-		Proof:  proof,
-	}
-
 	log.Lvl2(s.ServerIdentity(), "Got", len(proof), "proofs")
-	pl, err := protobuf.Encode(&req)
+	sb := s.db().GetByID(proof[0].View.ID)
+	req, err := viewchange.NewNewViewReq(sb.Roster, proof)
 	if err != nil {
-		log.Error("Couldn't encode request:", err)
-		return
-	}
-	if !s.verifyViewChange(req.Hash(), pl) {
-		log.Error("Will not ask for view change I cannot verify")
-		return
+		log.Errorf("couldn't create request: %v", err)
```

Comment:
Not bailing early?

Reply:
yes, correct. Converted the method to return an error, and handle them correctly.